x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/required_fields/required_fields_help_info.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useToggle } from 'react-use';
+import { EuiLink, EuiPopover, EuiText, EuiButtonIcon } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../../common/lib/kibana';
+
+/**
+ * Theme doesn't expose width variables. Using provided size variables will require
+ * multiplying it by another magic constant.
+ *
+ * 320px width looks
+ * like a [commonly used width in EUI](https://github.com/search?q=repo%3Aelastic%2Feui%20320&type=code).
+ */
+const POPOVER_WIDTH = 320;
+
+export function RequiredFieldsHelpInfo(): JSX.Element {
+  const [isPopoverOpen, togglePopover] = useToggle(false);
+  const { docLinks } = useKibana().services;
+
+  const button = (
+    <EuiButtonIcon
+      iconType="questionInCircle"
+      onClick={togglePopover}
+      aria-label="Open help popover"
+    />
+  );
+
+  return (
+    <EuiPopover button={button} isOpen={isPopoverOpen} closePopover={togglePopover}>
+      <EuiText style={{ width: POPOVER_WIDTH }} size="s">
+        <FormattedMessage
+          id="xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.fieldRequiredFieldsHelpText"
+          defaultMessage="Choose the fields that are needed for this rule to execute. You can read more about required fields in our {docLink}."
```

Comment:
Tightened up the popover text. I also suggest omitting the doc link, since we can probably tell the user all they need in the UI, and the docs won't have much more to say. Also we don't have a very targeted location to send users to: the [current destination](https://www.elastic.co/guide/en/security/master/rules-ui-create.html) is a really long topic, and the "required fields" step will be scattered across the page.
```suggestion
          defaultMessage="Choose the fields and data types needed for this rule to function. You can select the fields available in the rule's **Source** index patterns or data view."
```